hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/Processor.java

```diff
@@ -173,10 +173,11 @@ default boolean isCooperative() {
      * {@code complete()} returns {@code true}, this method won't be called
      * anymore.
      * <p>
-     * Processors must ensure they give a chance to the system to call this
-     * method. Especially, non-cooperative source processors must make sure not
-     * to spend too much time per invocation of {@code complete()}, as this
-     * will prolong the snapshotting phase and impede overall progress.
+     * Processors which override this method must ensure they give a chance
+     * to the system to call this method. Especially, non-cooperative
+     * source processors must make sure not to spend too much time per invocation
+     * of {@code complete()}, as this will prolong the snapshotting phase
+     * and impede overall progress.
```

Comment:
We can remove this paragraph altogether. The need to return from `complete()` is already documented there.